### PR TITLE
test/node-e2e: Update CoreOS update disabling

### LINKED ISSUE #31633: Failure Finished Host tmp-node-e2e-b6d45b2a-coreos-alpha-1122-0-0-v20160727 Test Suite
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/30321/node-pull-build-e2e-test/21737/

```
I0829 10:30:09.777786   22504 remote.go:210] Copying test artifacts from tmp-node-e2e-b6d45b2a-coreos-alpha-1122-0-0-v20160727
I0829 10:30:09.792249   22504 run_remote.go:559] Deleting instance "tmp-node-e2e-b6d45b2a-coreos-alpha-1122-0-0-v20160727"
================================================================
Failure Finished Host tmp-node-e2e-b6d45b2a-coreos-alpha-1122-0-0-v20160727 Test Suite
Running Suite: E2eNode Suite
```

Super unclear what happened here...

### PR DESCRIPTION
**Pull request by euank**
Previously in this saga... #25004

This disables update-engine and locksmithd with ignition instead of
cloud-init so that they're really totally 100% disabled. Our ignition guy promises.

Pretty much every way of disabling them with cloud-init is mildly racy.

Fixes #31633 

I think @vishh can say "I told you so" after the comment on https://github.com/kubernetes/kubernetes/pull/30023#discussion-diff-73431324 .. he was right, but it turns out "stop" there doesn't really work either because of the mess that is cloud-init. Fortunately, converting our cloud-init to json and calling it "ignition" works quite well :smile: 

Testing done: I ssh'd in and verified that yes, they're disabled. I didn't wait on the e2e tests to pass, so we'll let this PR check that.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31653)

### CODE REVIEW

**Comment by timstclair** [comment]
I don't understand why this works. Why does converting to JSON make a difference?

File: test/e2e_node/jenkins/coreos-init.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "name": "locksmithd.service",
+      "mask": true
+    },
+    {
+      "name": "docker.service",
+      "dropins": [{
+        "name": "10-disable-systemd-cgroup-driver.conf",
+        "contents": "[Service]\nCPUAccounting=yes\nMemoryAccounting=yes\nEnvironment=\"DOCKER_CGROUPS=\""
+      }]
+    }]
```
Comment (timstclair) [inline test/e2e_node/jenkins/coreos-init.json]: What about the `node-e2e-test-setup.service`? No longer needed?

File: test/e2e_node/jenkins/coreos-init.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (timstclair) [inline test/e2e_node/jenkins/coreos-init.json]: You don't need the users section anymore?

**Comment by Random-Liu** [comment]
@timstclair I've chatted with @yifan-gu (seems that @euank is not on slack currently).
The PR replace cloud-init with ignition, which runs in early boot process, which should be less racy (or no race at all?).

**Comment by yifan-gu** [comment]
@timstclair it's converting to ignition config instead of cloud config. Ignition runs so early that it provides better gurantees https://coreos.com/ignition/docs/latest/what-is-ignition.html https://coreos.com/ignition/docs/latest/boot-process.html

File: test/e2e_node/jenkins/coreos-init.json (hunk 1 above)
Comment (Random-Liu) [inline test/e2e_node/jenkins/coreos-init.json]: Same question here. :)

**Comment by timstclair** [comment]
Got it, thanks for the pointers. Inline questions still apply.

File: test/e2e_node/jenkins/coreos-init.json (hunk 1 above)
Comment (euank) [inline test/e2e_node/jenkins/coreos-init.json]: It was empty before, now it's gone. We'll add it back if/when we ever need it.

File: test/e2e_node/jenkins/coreos-init.json (hunk 2 above)
Comment (euank) [inline test/e2e_node/jenkins/coreos-init.json]: See `passwd.users` section, it contains the same basic information. Sorry for the re-ordering! I was going by the ignition spec for my order this time.

**Comment by timstclair** [comment]
LGTM

**Comment by Random-Liu** [comment]
LGTM, too! Thanks for the fixing! :)

**Pull request merged**